src/components/common/modals/ExchangeModal.vue

```diff
@@ -0,0 +1,295 @@
+        </el-input>
+        <span
+          v-if="_isError($v.exchangeForm.request_amount)"
+          class="el-form-item__error"
+        >{{ _showError($v.exchangeForm.request_amount) }}</span>
+      </el-form-item>
+      <span class="form-item-text">
+        Market price:
+        <span v-if="exchangeDialogRequestAsset && exchangeDialogOfferAsset && exchangeDialogPrice" class="form-item-text-amount">
+          1 {{ exchangeDialogOfferAsset }} ≈ {{ exchangeDialogPrice }} {{ exchangeDialogRequestAsset }}
+        </span>
+        <span v-else>...</span>
+      </span>
+      <el-form-item label="Counterparty" prop="to">
+        <el-input
+          v-model="$v.exchangeForm.to.$model"
+          placeholder="Account id"
+          :class="[
+            _isValid($v.exchangeForm.to) ? 'border_success' : '',
+            _isError($v.exchangeForm.to) ? 'border_fail' : ''
+          ]"
+        />
+        <span
+          v-if="_isError($v.exchangeForm.to)"
+          class="el-form-item__error"
+        >{{ _showError($v.exchangeForm.to) }}</span>
+      </el-form-item>
+      <el-form-item label="Additional information" prop="description">
+        <el-input
+          v-model="$v.exchangeForm.description.$model"
+          placeholder="Description"
+          resize="none"
+          :class="[
+            _isValid($v.exchangeForm.description) ? 'border_success' : '',
+            _isError($v.exchangeForm.description) ? 'border_fail' : ''
+          ]"
+        />
+        <span
+          v-if="_isError($v.exchangeForm.description)"
+          class="el-form-item__error"
+        >{{ _showError($v.exchangeForm.description) }}</span>
+      </el-form-item>
+    </el-form>
+    <el-button
+      class="dialog-form_buttons fullwidth action"
+      @click="onSubmitExchangeDialog()"
+      style="margin-top: 40px"
+      :loading="isExchangeSending"
+    >
+      EXCHANGE
+    </el-button>
+  </el-dialog>
+</template>
+
+<script>
+import { mapGetters, mapActions } from 'vuex'
+
+import numberFormat from '@/components/mixins/numberFormat'
+import messageMixin from '@/components/mixins/message'
+import NOTIFICATIONS from '@/data/notifications'
+
+import {
+  _user,
+  _amount,
+  _wallet,
+  errorHandler
+} from '@/components/mixins/validation'
+import { required, maxLength } from 'vuelidate/lib/validators'
+
+export default {
+  name: 'exchange-modal',
+  mixins: [
+    messageMixin,
+    numberFormat,
+    errorHandler
+  ],
+  validations () {
+    const wallet = this.wallets.find(x => x.asset === this.exchangeDialogOfferAsset)
+    const { amount, precision } = wallet || { amount: 0, precision: 0 }
+    return {
+      exchangeForm: {
+        to: {
+          required,
+          _userDomain: _user.nameDomain,
+          _userExist: _user.nameExist
+        },
+        request_amount: {
+          required,
+          _asset: _wallet.asset(this.exchangeDialogRequestAsset),
+          _amount: _amount({ amount, precision })
```

Comment:
It's not correct, because we can't limit request amount. Now we limit with offer amount.

Reply:
Just a comment, but limits hasn't validation.